### PR TITLE
DOC: add examples to docstrings of assert_ functions

### PR DESCRIPTION
**Pull request by nguevara**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/_testing.py [hunk 1]
```diff
@@ -739,6 +739,12 @@ def assert_index_equal(
     obj : str, default 'Index'
         Specify object name being compared, internally used to show appropriate
         assertion message.
+    Examples
```
Comment (MarcoGorelli) [inline pandas/_testing.py:742]: can you add a new line before `Examples`?

File: pandas/_testing.py [hunk 2]
```diff
@@ -739,6 +739,12 @@ def assert_index_equal(
...
+    --------
+    >>> from pandas._testing import assert_index_equal
+    >>> a=pd.Index([1,2,3])
```
Comment (MarcoGorelli) [inline pandas/_testing.py:745]: can you add spaces around the `=`? so `a = pd.Index`...

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks @nguevara

File: pandas/_testing.py [hunk 3]
```diff
@@ -739,6 +739,12 @@ def assert_index_equal(
     obj : str, default 'Index'
         Specify object name being compared, internally used to show appropriate
         assertion message.
+    Examples
+    --------
+    >>> from pandas._testing import assert_index_equal
```
Comment (jorisvandenbossche) [inline pandas/_testing.py:744]: The public import is `pandas.testing`, can you use that?

File: pandas/_testing.py [hunk 4]
```diff
@@ -1334,6 +1348,13 @@ def assert_series_equal(
     obj : str, default 'Series'
         Specify object name being compared, internally used to show appropriate
         assertion message.
+
+    Examples
+    --------
+    >>> from pandas.testing import assert_series_equal
+    >>> a = pd.Series([1,2,3,4])
+    >>> b = pd.Series([1,2,3,4])
+    >>> assert_series_equal(a,b)
```
Comment (MarcoGorelli) [inline pandas/_testing.py:1357]:
```suggestion
    >>> a = pd.Series([1, 2, 3, 4])
    >>> b = pd.Series([1, 2, 3, 4])
    >>> assert_series_equal(a, b)
```

File: pandas/_testing.py [hunk 5]
```diff
@@ -1205,6 +1212,13 @@ def assert_extension_array_equal(
     Missing values are checked separately from valid values.
     A mask of missing values is computed for each and checked to match.
     The remaining all-valid values are cast to object dtype and checked.
+
+    Examples
+    --------
+    >>> from pandas.testing import assert_extension_array_equal
+    >>> a = pd.Series([1,2,3,4])
+    >>> b,c = a.array,a.array
+    >>> assert_extension_array_equal(b,c)
```
Comment (MarcoGorelli) [inline pandas/_testing.py:1221]:
```suggestion
    >>> a = pd.Series([1, 2, 3, 4])
    >>> b, c = a.array, a.array
    >>> assert_extension_array_equal(b, c)
```

File: pandas/_testing.py [hunk 6]
```diff
@@ -739,6 +739,13 @@ def assert_index_equal(
     obj : str, default 'Index'
         Specify object name being compared, internally used to show appropriate
         assertion message.
+
+    Examples
+    --------
+    >>> from pandas.testing import assert_index_equal
+    >>> a = pd.Index([1,2,3])
+    >>> b = pd.Index([1,2,3])
+    >>> assert_index_equal(a,b)
```
Comment (MarcoGorelli) [inline pandas/_testing.py:748]:
```suggestion
    >>> a = pd.Index([1, 2, 3])
    >>> b = pd.Index([1, 2, 3])
    >>> assert_index_equal(a, b)
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks @nguevara ! Just making a nitpick comment to make the style more consistent with the rest of the codebase

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks a lot @nguevara !

**Pull request merged**